Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -480,6 +480,11 @@ public void incrementLoyaltyActivationsAvailable(int max) {
         }
     }
 
+    @Override
+    public void setLoyaltyActivationsAvailable(int loyaltyActivationsAvailable) {
+        this.loyaltyActivationsAvailable = loyaltyActivationsAvailable;
+    }
```

Comment:
It does do the same thing as incrementLoyaltyActiviationsAvailable(2) but anything more than 2 you would need to call incrementLoyaltyActiviationsAvailable multiple times, if thats the desired behavior then I can switch to using that. Just thought this way would be easier.

Reply:
Yeah, I do prefer your implementation for this type of effect and believe that setting to 2 is more intuitive than incrementing to 2. I was just reading the way xmage handles adding activations to a planeswalker's abilities and found that [[The Chain Veil]]'s implementation has some issues, mainly that it simply increments the number of possible activations by one. This means that activating The Chain Veil on the pre-combat main phase increases the number of Kaito's activations to two, then during combat when Kaito's ability activates it will set the number to two, effectively doing nothing, when I believe the correct interaction should be that Kaito's loyalty abilities should be activatable 3 times that turn.

Not a problem with your implementation, just an issue with the Chain Veil I noticed while reading the code. Could you update your function to:
```java
@Override
public void setLoyaltyActivationsAvailable(int setActivations) {
    if (this.loyaltyActivationsAvailable < setActivations) {
        this.loyaltyActivationsAvailable = setActivations;
    }
}
```
Then change UrzaPlaneswalker, OathOfTeferi, and UrzaAssemblesTheTitans to use your updated version. The original function can stick around until a better way of implementing The Chain Veil is figured out

Reply:
👍